test/housekeeping_test.js

```diff
@@ -323,5 +350,92 @@ describe('House Keeper', () => {
       });
     }
   });
-});
 
+  it('should call sweepVolumes exactly once', async() => {
+    houseKeeperMock.expects("sweepVolumes").exactly(regions.length);
+    
+    describeInstancesStub.returns({
+      Reservations: [
+      ]
+    });
+    
+    describeSpotInstanceRequestsStub.returns({
+      SpotInstanceRequests: [
+      ]
+    });
+    
+    await houseKeeper.sweep();
+    houseKeeperMock.verify();
+  });
+
+  it('should not fail if no volume data is returned', async() => {
+    houseKeeperMock.expects("handleVolumeData").never();
+    describeInstancesStub.returns({
+      Reservations: [
+      ]
+    });
+    
+    describeSpotInstanceRequestsStub.returns({
+      SpotInstanceRequests: [
+      ]
+    });
+    
+    describeVolumesStub.returns({
+      Volumes: [
+      ]
+    });
+
+    await houseKeeper.sweep();
+    houseKeeperMock.verify();
+  });
+  
+  it('should call handleVolumeData once per volume', async() => {
+    houseKeeperMock.expects("handleVolumeData").twice();
+      
+    describeInstancesStub.returns({
+      Reservations: [
+      ]
+    });
+    
+    describeSpotInstanceRequestsStub.returns({
+      SpotInstanceRequests: [
+      ]
+    });
+
+    describeVolumesStub.withArgs(ec2[region], 'describeVolumes', houseKeeper.describeVolumesParams)
```

Comment:
It seems like exposing the `describeVolumesParams` is only needed in order to use here for matching the call.  Maybe instead, we might be able to use the Sinon [matcher api](http://sinonjs.org/releases/v4.0.1/matchers/)?

Reply:
I came up with a solution that is a bit better, I think, but still not perfect. Sinon supports custom matchers, which return true or false to indicate whether the input matches or not. I tested using a custom matcher, and it seems that only the first parameter gets passed along. Luckily we only need to check the value of the first parameter, so it works, but it seems a bit odd.

Reply:
I have to look at the code to fully understand, but it might be that there's a 1:1 relationship between arguments and matchers?  I'll look at the code